In [29]:
#Problem 1
import numpy as np
from scipy.stats import norm
from scipy.stats import t

c = .90 #confidence interval

data = [-23, -16, -3, -9, -1, 13, -16, 15, 31, -22, -14, -25]
dataLength = len(data)
dataMean = np.mean(data)
print("The sample mean of the data is " + str(dataMean))
SE = (np.std(data, ddof=(1))) / np.sqrt(dataLength)
print("The standard error of the data is " + str(SE))

t_c = t.ppf(1 - (1 - c)/2, df = (dataLength -1))
print("The t score for this data set at at a confidence interval of {} is {}.".format(c,t_c))
lower = dataMean - (t_c * SE)
upper = dataMean + (t_c * SE)
print("The interval of this data for a confidence interval of {} is from {} to {}".format(c, lower, upper))



The sample mean of the data is -5.833333333333333
The standard error of the data is 5.062987100895912
The t score for this data set at at a confidence interval of 0.9 is 1.7958848187036691.
The interval of this data for a confidence interval of 0.9 is from -14.925875005124801 to 3.2592083384581363


Problem 2: Confidence Intervals
1. For this test I would use a t-test because of the smaller size of the sample.
The sample mean of the data is -5.833333333333333
The standard error of the data is 5.062987100895912
The t score for this data set at at a confidence interval of 0.9 is 1.7958848187036691.
The interval of this data for a confidence interval of 0.9 is from -14.925875005124801 to  3.2592083384581363

In [16]:
#Problem 2
import numpy as np
from scipy.stats import norm
from scipy.stats import t

c = .95 #confidence interval

data = [-23, -16, -3, -9, -1, 13, -16, 15, 31, -22, -14, -25]
dataLength = len(data)
dataMean = np.mean(data)
print("The sample mean of the data is " + str(dataMean))
SE = (np.std(data, ddof=(1))) / np.sqrt(dataLength)
print("The standard error of the data is " + str(SE))

t_c = t.ppf(1 - (1 - c)/2, df = (dataLength - 1))
print("The t score for this data set at at a confidence interval of {} is {}.".format(c,t_c))
lower = dataMean - (t_c * SE)
upper = dataMean + (t_c * SE)
print("The interval of this data for a confidence interval of {} is from {} to {}".format(c, lower, upper))


The sample mean of the data is -5.833333333333333
The standard error of the data is 16.792028532080998
The t score for this data set at at a confidence interval of 0.95 is 2.200985160082949.
The interval of this data for a confidence interval of 0.95 is from -42.79233894013308 to 31.12567227346641


Problem 2: Confidence Intervals
2. 
The sample mean of the data is -5.833333333333333
The standard error of the data is 5.062987100895912
The t score for this data set at at a confidence interval of 0.95 is 2.200985160082949.
The interval of this data for a confidence interval of 0.95 is from -16.976892808096625 to 5.31022614142996

With a confidence interval of 95%, it appears as though our t score has increased and our interval has become wider. The standard error and mean remained the same in both conditions.

In [11]:
#Problem 3
import numpy as np
from scipy.stats import norm
from scipy.stats import t
sd = 16.836
c = .95 #confidence interval

data = [-23, -16, -3, -9, -1, 13, -16, 15, 31, -22, -14, -25]
dataLength = len(data)
dataMean = np.mean(data)
print("The sample mean of the data is " + str(dataMean))
SE = (sd) / np.sqrt(dataLength)
print("The standard error of the data is " + str(SE))

t_c = t.ppf(1 - (1 - c)/2, df = (dataLength - 1))
print("The t score for this data set at at a confidence interval of {} is {}.".format(c,t_c))
lower = dataMean - (t_c * SE)
upper = dataMean + (t_c * SE)
print("The interval of this data for a confidence interval of {} is from {} to {}".format(c, lower, upper))


The sample mean of the data is -5.833333333333333
The standard error of the data is 4.860134566038269
The t score for this data set at at a confidence interval of 0.95 is 2.200985160082949.
The interval of this data for a confidence interval of 0.95 is from -16.530417389189747 to 4.86375072252308


Problem 2: Confidence Intervals
3.
Due to the sample size still being less than 30, we will still use a t test to perform this experiement, with the added difference of using our newfound standard deviation to calculate our standard error. 
This leaves us with the following data:
The sample mean of the data is -5.833333333333333
The standard error of the data is 4.860134566038269
The t score for this data set at at a confidence interval of 0.95 is 2.200985160082949.
The interval of this data for a confidence interval of 0.95 is from -16.530417389189747 to 4.86375072252308

With this new standard deviation, our confidence interval has shrunk as a result of our smaller standard error in comparison to problem 2, but it is still larger than problem 1. The t score and mean did not change with a new standard deviation.

In [41]:
#Problem 4
import numpy as np
from scipy.stats import norm
from scipy.stats import t
#looking for upper to be equal to 0
upper = 0

data = [-23, -16, -3, -9, -1, 13, -16, 15, 31, -22, -14, -25]
dataLength = len(data)
dataMean = np.mean(data)
print("The sample mean of the data is " + str(dataMean))
SE = (np.std(data, ddof=(1))) / np.sqrt(dataLength)
print("The standard error of the data is " + str(SE))
t_c = (upper - dataMean) / SE
print("The T score of the data is " + str(t_c))
#t_c = t.ppf(1 - (1 - c)/2, df = (dataLength - 1))
#print("The t score for this data set at at a confidence interval of {} is {}.".format(c,t_c))
#lower = dataMean - (t_c * SE)
#upper = dataMean + (t_c * SE)
#print("The interval of this data for a confidence interval of {} is from {} to {}".format(c, lower, upper))
#p = t.cdf(t_c, df = (dataLength - 1))
p = t.cdf(-abs(t_c), df = (dataLength - 1))
p = 1 - p
print("The probability that the team will lose is " + str(p))
print("We can say with 86.31582571446504% confidence that this team will lose on average.")

The sample mean of the data is -5.833333333333333
The standard error of the data is 5.062987100895912
The T score of the data is 1.152152517295789
The probability that the team will lose is 0.8631582571446504


Problem 2: Confidence Intervals
4. We can say with 86.31582571446504% confidence that this team will lose on average.